Creating views with database_to_upper=false fails.

Using `DATABASE_TO_UPPER=FALSE`, it is not possible to create the bellow view.  It seems  the column name is wrongly computed.


```java
package org.h2;
import org.h2.jdbcx.JdbcDataSource;

import java.sql.Connection;
import java.sql.Statement;


public class Didier {

    public static void main(String[] args) throws Exception {
        JdbcDataSource ds = new JdbcDataSource();
        ds.setURL("jdbc:h2:mem:;DATABASE_TO_UPPER=FALSE");
        Connection conn = ds.getConnection();

        Statement statement = conn.createStatement();
        statement.execute("CREATE TABLE \"a table\" (a VARCHAR, b INT);");  // important, the table name contains a space
        statement.execute("CREATE VIEW foo AS SELECT * FROM \"a table\";");
        statement.execute("SELECT * FROM foo");
    }

}
```

Error message

```
/home/didier/opt/jdks/jdk1.8.0_191/bin/java -javaagent:/home/didier/opt/idea-IU-182.5107.16/lib/idea_rt.jar=33309:/home/didier/opt/idea-IU-182.5107.16/bin -Dfile.encoding=UTF-8 -classpath /home/didier/opt/jdks/jdk1.8.0_191/jre/lib/charsets.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/deploy.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/cldrdata.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/dnsns.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/jaccess.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/jfxrt.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/localedata.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/nashorn.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/sunec.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/sunjce_provider.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/sunpkcs11.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/ext/zipfs.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/javaws.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/jce.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/jfr.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/jfxswt.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/jsse.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/management-agent.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/plugin.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/resources.jar:/home/didier/opt/jdks/jdk1.8.0_191/jre/lib/rt.jar:/home/didier/dev/h2database/h2/target/classes:/home/didier/.m2/repository/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar:/home/didier/.m2/repository/org/apache/lucene/lucene-core/5.5.5/lucene-core-5.5.5.jar:/home/didier/.m2/repository/org/apache/lucene/lucene-analyzers-common/5.5.5/lucene-analyzers-common-5.5.5.jar:/home/didier/.m2/repository/org/apache/lucene/lucene-queryparser/5.5.5/lucene-queryparser-5.5.5.jar:/home/didier/.m2/repository/org/apache/lucene/lucene-queries/5.5.5/lucene-queries-5.5.5.jar:/home/didier/.m2/repository/org/apache/lucene/lucene-sandbox/5.5.5/lucene-sandbox-5.5.5.jar:/home/didier/.m2/repository/org/slf4j/slf4j-api/1.6.0/slf4j-api-1.6.0.jar:/home/didier/.m2/repository/org/osgi/org.osgi.core/4.2.0/org.osgi.core-4.2.0.jar:/home/didier/.m2/repository/org/osgi/org.osgi.enterprise/4.2.0/org.osgi.enterprise-4.2.0.jar:/home/didier/.m2/repository/org/locationtech/jts/jts-core/1.15.0/jts-core-1.15.0.jar:/home/didier/opt/jdks/jdk1.8.0_191/lib/tools.jar org.h2.Didier
Exception in thread "main" org.h2.jdbc.JdbcSQLSyntaxErrorException: View "PUBLIC.""foo""" is invalid: "Column ""a"" not found [42122-197]"; SQL statement:
SELECT * FROM foo [90109-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:573)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	at org.h2.message.DbException.get(DbException.java:194)
	at org.h2.table.TableView.getScanIndex(TableView.java:482)
	at org.h2.table.TableFilter.getBestPlanItem(TableFilter.java:210)
	at org.h2.table.Plan.calculateCost(Plan.java:120)
	at org.h2.command.dml.Optimizer.testPlan(Optimizer.java:181)
	at org.h2.command.dml.Optimizer.calculateBestPlan(Optimizer.java:82)
	at org.h2.command.dml.Optimizer.optimize(Optimizer.java:240)
	at org.h2.command.dml.Select.preparePlan(Select.java:1412)
	at org.h2.command.dml.Select.prepare(Select.java:1278)
	at org.h2.command.Parser.prepareCommand(Parser.java:682)
	at org.h2.engine.Session.prepareLocal(Session.java:625)
	at org.h2.engine.Session.prepareCommand(Session.java:563)
	at org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)
	at org.h2.Didier.main(Didier.java:18)
Caused by: org.h2.message.DbException: Column "a" not found [42122-197]
	at org.h2.message.DbException.get(DbException.java:205)
	at org.h2.message.DbException.get(DbException.java:181)
	at org.h2.expression.ExpressionColumn.getColumnException(ExpressionColumn.java:184)
	at org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:166)
	at org.h2.command.dml.Select.prepare(Select.java:1257)
	at org.h2.command.Parser.prepare(Parser.java:661)
	at org.h2.engine.Session.prepare(Session.java:590)
	at org.h2.table.TableView.compileViewQuery(TableView.java:121)
	at org.h2.table.TableView.initColumnsAndTables(TableView.java:173)
	at org.h2.table.TableView.init(TableView.java:114)
	at org.h2.table.TableView.<init>(TableView.java:73)
	at org.h2.command.ddl.CreateView.update(CreateView.java:125)
	at org.h2.command.CommandContainer.update(CommandContainer.java:133)
	at org.h2.command.Command.executeUpdate(Command.java:267)
	at org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:233)
	at org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)
	at org.h2.Didier.main(Didier.java:17)
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Column "a" not found; SQL statement:
CREATE FORCE VIEW PUBLIC."foo" AS
SELECT
    a table."a",
    a table."b"
FROM PUBLIC."a table" [42122-197]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:451)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:427)
	... 17 more

Process finished with exit code 1
```


tested on:  1c185f1b1c739185dfc20eb01673fcef15475a25

--- a/h2/src/docsrc/html/changelog.html
+++ b/h2/src/docsrc/html/changelog.html
@@ -21,6 +21,8 @@ Change Log
 
 <h2>Next Version (unreleased)</h2>
 <ul>
+<li>Issue #1735: Creating views with DATABASE_TO_UPPER=FALSE fails
+</li>
 <li>Issue #1732: source.html does not work
 </li>
 <li>Issue #1730: Show error in H2 Console if specified driver is not compatible with URL

--- a/h2/src/main/org/h2/expression/ExpressionColumn.java
+++ b/h2/src/main/org/h2/expression/ExpressionColumn.java
@@ -59,22 +59,11 @@ public class ExpressionColumn extends Expression {
 
     @Override
     public StringBuilder getSQL(StringBuilder builder) {
-        boolean quote = database.getSettings().databaseToUpper;
         if (schemaName != null) {
-            if (quote) {
-                Parser.quoteIdentifier(builder, schemaName);
-            } else {
-                builder.append(schemaName);
-            }
-            builder.append('.');
+            Parser.quoteIdentifier(builder, schemaName).append('.');
         }
         if (tableAlias != null) {
-            if (quote) {
-                Parser.quoteIdentifier(builder, tableAlias);
-            } else {
-                builder.append(tableAlias);
-            }
-            builder.append('.');
+            Parser.quoteIdentifier(builder, tableAlias).append('.');
         }
         if (column != null) {
             if (derivedName != null) {
@@ -82,10 +71,8 @@ public class ExpressionColumn extends Expression {
             } else {
                 builder.append(column.getSQL());
             }
-        } else if (quote) {
-            Parser.quoteIdentifier(builder, columnName);
         } else {
-            builder.append(columnName);
+            Parser.quoteIdentifier(builder, columnName);
         }
         return builder;
     }

--- a/h2/src/test/org/h2/test/db/TestCompatibility.java
+++ b/h2/src/test/org/h2/test/db/TestCompatibility.java
@@ -117,6 +117,13 @@ public class TestCompatibility extends TestDb {
         assertEquals(1000000000000000000000e10, rs.getDouble(2));
         assertEquals(0xfafbL, rs.getLong(3));
         assertFalse(rs.next());
+
+        stat.execute("create table \"t 1\" (a int, b int)");
+        stat.execute("create view v as select * from \"t 1\"");
+        stat.executeQuery("select * from v").close();
+        stat.execute("drop view v");
+        stat.execute("drop table \"t 1\"");
+
         c.close();
     }
 

